src/directives/badge/badge.directive.ts

```diff
@@ -0,0 +1,286 @@
+                ) {
+                    this._truncatedText = true;
+                    finalText = `${changes.content.currentValue.substr(
+                        0,
+                        this.maxValue - 1
+                    )}&hellip;`;
+                } else {
+                    this._truncatedText = false;
+                }
+            }
+
+            this._displayText = finalText;
+        }
+
+        // if text is only set, set aria label from it
+        if (changes.content && !this.ariaLabel) {
+            this.ariaLabel = changes.content.currentValue;
+        }
+
+        // force update to text color if background color changes and text color not explicitly set
+        if (changes.backgroundColor && this._textColor !== this._lightColor) {
+            this.textColor = this._textColor.toHex();
+        }
+
+        this.updateBadge();
+    }
+
+    ngOnDestroy(): void {
+        if (this._renderer.destroyNode) {
+            this._renderer.destroyNode(this._badgeElement);
+        }
+    }
+
+    private updateBadge(): HTMLElement {
+        const badgeElement: HTMLSpanElement = this._renderer.createElement(
+            'span'
+        );
+        this.clearExisting(this._class);
+
+        badgeElement.classList.add(this._class);
+        badgeElement.classList.add('hidden');
+
+        badgeElement.innerHTML = this._displayText;
+        badgeElement.style.setProperty('color', this._textColor.toHex());
+
+        if (this.ariaLabel) {
+            badgeElement.setAttribute('aria-label', this.ariaLabel);
+        }
+
+        if (this._truncatedText) {
+            badgeElement.setAttribute('title', this._content);
+        }
+
+        if (this._backgroundColor) {
+            badgeElement.style.setProperty(
+                'background',
+                this._backgroundColor.toHex()
+            );
+        }
+
+        this._badgeElement = badgeElement;
+        this._renderer.appendChild(
+            this._element.nativeElement,
+            this._badgeElement
+        );
+
+        if (!this.inline) {
+            const badgeWidth = this._badgeElement.offsetWidth;
+
+            if (badgeWidth) {
+                const xPos = this.overlap
+                    ? badgeWidth / 2
+                    : badgeWidth + this._noOverlapOffset;
+
+                this._badgeElement.style.setProperty('right', `-${xPos}px`);
+            }
+        }
+
+        badgeElement.classList.remove('hidden');
+
+        return this._badgeElement;
+    }
+
+    private clearExisting(className: string): void {
+        const parentElement: HTMLElement = this._element.nativeElement;
+        let count: number = parentElement.children.length;
+
+        while (count--) {
+            const current = parentElement.children[count];
+
+            if (current.classList.contains(className)) {
+                parentElement.removeChild(current);
+            }
+        }
+    }
+
+    private _determineContrastColor(): ThemeColor {
```

Comment:
Just to make you aware of our function naming conventions as they may not be totally obvious at first. Function names only get an underscore if they should be private but can't be marked private because they are used in an Angular template. This is our way to indicate that a public function should not be used. Otherwise we can omit the underscore